src/main/scala/org/broadinstitute/hail/driver/Grep.scala

```diff
@@ -34,11 +34,19 @@ object Grep extends Command {
     val regex = args.head.r
     val files = args.tail
 
-    sc.union(files.map { f =>
-      sc.textFile(f)
-    }).filter(line => regex.findFirstIn(line).isDefined)
+    sc.textFilesLines(files.toArray)
+      .filter(line => regex.findFirstIn(line.value).isDefined)
       .take(options.max)
-      .foreach(println)
+      .groupBy(_.source.asInstanceOf[TextContext].file)
+      .foreach { case (file, lines) =>
+        println(s"$file:")
+        log.info(s"$file:")
+        lines.foreach { line =>
+          val lineToPrint = truncate(line.value, 60)
+          log.info("\t" + line.value)
+          println(s"\t$lineToPrint")
+        }
```

Comment:
In GNU grep, `-H` turns on filename printing and it looks like:
```bash
src/test/scala/org/broadinstitute/hail/variant/KuduUtilsSuite.scala:        5, 6L, 7.toFloat, 8.0, "a",
src/test/scala/org/broadinstitute/hail/variant/vsm/VSMSuite.scala:    val master = "quickstart.cloudera"
```
would it be worth replicating the style?